TODO

```diff
@@ -0,0 +1,8 @@
+datastructures/core/executiongraph.py
```

Comment:
Are theses TODOs completed? If these need to be addressed in a new PR can we make an issue for these?

Reply:
The only TODOs that are from me are in utils.py. I will take a look.